uxarray/core/dataset.py

```diff
@@ -0,0 +1,138 @@
+import numpy as np
+import xarray as xr
+
+from collections.abc import Hashable
+
+from typing import Optional
+
+from uxarray.core.dataarray import UxDataArray
+from uxarray.core.grid import Grid
+
+
+class UxDataset(xr.Dataset):
+
+    _uxgrid = None
+
+    def __init__(self,
```

Comment:
@headtr1ck if you have the time, your comments on subclassing here  (also `dataarray.py`) would be valuable.

Reply:
we received some invaluable comments from @headtr1ck at the time; resolving this now